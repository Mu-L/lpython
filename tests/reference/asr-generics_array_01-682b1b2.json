{
    "basename": "asr-generics_array_01-682b1b2",
    "cmd": "lpython --show-asr --no-color {infile} -o {outfile}",
    "infile": "tests/../integration_tests/generics_array_01.py",
    "infile_hash": "6e943dd0e26ab4d1ffb6ed6909a365b4135b6f5295957b2478cfb479",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-generics_array_01-682b1b2.stdout",
    "stdout_hash": "d1aee82339e69feb96790f353c125ef8a82b4d7e37ad0e716b8d3210",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}