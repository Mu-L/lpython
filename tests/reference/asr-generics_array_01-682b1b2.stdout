(TranslationUnit
    (SymbolTable
        1
        {
            __main__:
                (Module
                    (SymbolTable
                        2
                        {
                            T:
                                (Variable
                                    2
                                    T
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (TypeParameter
                                        T
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                    .false.
                                    .false.
                                    ()
                                    .false.
                                    .false.
                                ),
                            __asr_generic_f_0:
                                (Function
                                    (SymbolTable
                                        212
                                        {
                                            _lpython_return_variable:
                                                (Variable
                                                    212
                                                    _lpython_return_variable
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    212
                                                    i
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            lst:
                                                (Variable
                                                    212
                                                    lst
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 4)
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    __asr_generic_f_0
                                    (FunctionType
                                        [(Array
                                            (Integer 4)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                        (Integer 4)]
                                        (Integer 4)
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 212 lst)
                                    (Var 212 i)]
                                    [(Assignment
                                        (ArrayItem
                                            (Var 212 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4) Decimal)
                                            ())]
                                            (Integer 4)
                                            RowMajor
                                            ()
                                        )
                                        (Var 212 i)
                                        ()
                                        .false.
                                    )
                                    (Assignment
                                        (Var 212 _lpython_return_variable)
                                        (ArrayItem
                                            (Var 212 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4) Decimal)
                                            ())]
                                            (Integer 4)
                                            RowMajor
                                            ()
                                        )
                                        ()
                                        .false.
                                    )
                                    (Return)]
                                    (Var 212 _lpython_return_variable)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            __main__global_stmts:
                                (Function
                                    (SymbolTable
                                        213
                                        {
                                            
                                        })
                                    __main__global_stmts
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [use_array]
                                    []
                                    [(SubroutineCall
                                        2 use_array
                                        ()
                                        []
                                        ()
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            f:
                                (Function
                                    (SymbolTable
                                        210
                                        {
                                            _lpython_return_variable:
                                                (Variable
                                                    210
                                                    _lpython_return_variable
                                                    []
                                                    ReturnVar
                                                    ()
                                                    ()
                                                    Default
                                                    (TypeParameter
                                                        T
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            i:
                                                (Variable
                                                    210
                                                    i
                                                    []
                                                    In
                                                    ()
                                                    ()
                                                    Default
                                                    (TypeParameter
                                                        T
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            lst:
                                                (Variable
                                                    210
                                                    lst
                                                    []
                                                    InOut
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (TypeParameter
                                                            T
                                                        )
                                                        [(()
                                                        ())]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    f
                                    (FunctionType
                                        [(Array
                                            (TypeParameter
                                                T
                                            )
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                        (TypeParameter
                                            T
                                        )]
                                        (TypeParameter
                                            T
                                        )
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    []
                                    [(Var 210 lst)
                                    (Var 210 i)]
                                    [(Assignment
                                        (ArrayItem
                                            (Var 210 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4) Decimal)
                                            ())]
                                            (TypeParameter
                                                T
                                            )
                                            RowMajor
                                            ()
                                        )
                                        (Var 210 i)
                                        ()
                                        .false.
                                    )
                                    (Assignment
                                        (Var 210 _lpython_return_variable)
                                        (ArrayItem
                                            (Var 210 lst)
                                            [(()
                                            (IntegerConstant 0 (Integer 4) Decimal)
                                            ())]
                                            (TypeParameter
                                                T
                                            )
                                            RowMajor
                                            ()
                                        )
                                        ()
                                        .false.
                                    )
                                    (Return)]
                                    (Var 210 _lpython_return_variable)
                                    Public
                                    .false.
                                    .false.
                                    ()
                                ),
                            use_array:
                                (Function
                                    (SymbolTable
                                        211
                                        {
                                            array:
                                                (Variable
                                                    211
                                                    array
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Array
                                                        (Integer 4)
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 1 (Integer 4) Decimal))]
                                                        FixedSizeArray
                                                    )
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                ),
                                            x:
                                                (Variable
                                                    211
                                                    x
                                                    []
                                                    Local
                                                    ()
                                                    ()
                                                    Default
                                                    (Integer 4)
                                                    ()
                                                    Source
                                                    Public
                                                    Required
                                                    .false.
                                                    .false.
                                                    .false.
                                                    ()
                                                    .false.
                                                    .false.
                                                )
                                        })
                                    use_array
                                    (FunctionType
                                        []
                                        ()
                                        Source
                                        Implementation
                                        ()
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        .false.
                                        []
                                        .false.
                                    )
                                    [__asr_generic_f_0]
                                    []
                                    [(Assignment
                                        (Var 211 array)
                                        (ArrayBroadcast
                                            (ArrayItem
                                                (Var 211 array)
                                                [(()
                                                (IntegerConstant 0 (Integer 4) Decimal)
                                                ())]
                                                (Integer 4)
                                                RowMajor
                                                ()
                                            )
                                            (ArrayConstant
                                                4
                                                [1]
                                                (Array
                                                    (Integer 4)
                                                    [((IntegerConstant 1 (Integer 4) Decimal)
                                                    (IntegerConstant 1 (Integer 4) Decimal))]
                                                    FixedSizeArray
                                                )
                                                ColMajor
                                            )
                                            (Array
                                                (Integer 4)
                                                [((IntegerConstant 0 (Integer 4) Decimal)
                                                (IntegerConstant 1 (Integer 4) Decimal))]
                                                FixedSizeArray
                                            )
                                            ()
                                        )
                                        ()
                                        .false.
                                    )
                                    (Assignment
                                        (Var 211 x)
                                        (IntegerConstant 69 (Integer 4) Decimal)
                                        ()
                                        .false.
                                    )
                                    (Print
                                        (StringFormat
                                            ()
                                            [(FunctionCall
                                                2 __asr_generic_f_0
                                                ()
                                                [((ArrayPhysicalCast
                                                    (Var 211 array)
                                                    FixedSizeArray
                                                    DescriptorArray
                                                    (Array
                                                        (Integer 4)
                                                        [((IntegerConstant 0 (Integer 4) Decimal)
                                                        (IntegerConstant 1 (Integer 4) Decimal))]
                                                        DescriptorArray
                                                    )
                                                    ()
                                                ))
                                                ((Var 211 x))]
                                                (Integer 4)
                                                ()
                                                ()
                                            )]
                                            FormatPythonFormat
                                            (String 1 () ExpressionLength CString)
                                            ()
                                        )
                                    )]
                                    ()
                                    Public
                                    .false.
                                    .false.
                                    ()
                                )
                        })
                    __main__
                    [numpy]
                    ()
                    .false.
                    .false.
                ),
            lpython_builtin:
                (IntrinsicModule lpython_builtin),
            main_program:
                (Program
                    (SymbolTable
                        214
                        {
                            __main__global_stmts:
                                (ExternalSymbol
                                    214
                                    __main__global_stmts
                                    2 __main__global_stmts
                                    __main__
                                    []
                                    __main__global_stmts
                                    Public
                                )
                        })
                    main_program
                    [__main__]
                    [(SubroutineCall
                        214 __main__global_stmts
                        2 __main__global_stmts
                        []
                        ()
                    )]
                ),
            numpy:
                (Module numpy)
        })
    []
)
